### PR TITLE
Support custom array of methods

### PR DESCRIPTION
**Pull request by YerkoPalma**
use the same handler for DELETE, GET, PATCH, PUT and POST methods when '*' is specified
related: [merry/93](https://github.com/shipharbor/merry/issues/93)

### CODE REVIEW

**Comment by YerkoPalma** [comment]
Mmm.. was just thinking that it might be better to handle a custom set of methods with the same handler. For example, if I have some common behaviour for PUT and PATCH method, it would be handy to do

```js
router.route(['PUT', 'PATCH'], '/foo', function (req, res, ctx) {
  // ...
})
```

Basically, use the behaviour in this PR when the method is an array of strings. What do you think? @yoshuawuyts

**Comment by yoshuawuyts** [comment]
@YerkoPalma think you're right - actually providing an array of methods might be nicer. Don't think we'd still need the `'*'` handler then? - or do you like it as a shorthand? My concern is that `'*'` could be confused to handle _all_ routes, where it actually only handles a common subsection; debugging that could be tricky

**Comment by YerkoPalma** [comment]
Yeah, I feel like the array support is better than the '*' handler, also because of some methods, like HEAD, that you only need to handle in very specific scenarios. Will update the PR :)

**Comment by YerkoPalma** [comment]
@yoshuawuyts any update on this?

File: test.js [hunk 1]
```diff
@@ -107,4 +107,43 @@ tape('server-router', function (t) {
     var server = http.createServer(router.start()).listen()
     http.get(`http://localhost:${getPort(server)}/hello`)
   })
+
+  t.test('should handle \'*\' method', function (t) {
```
Comment (yoshuawuyts) [inline test.js]: This test should be renamed I reckon

Comment (YerkoPalma) [reply test.js]: Done

**Review by yoshuawuyts — APPROVED** [review]
Thanks!

**Comment by yoshuawuyts** [comment]
`v5.2.0` :tada:

**Pull request merged**